src/javascript/expression.rs

```diff
@@ -326,6 +334,34 @@ impl<'module> Generator<'module> {
             .append(" ")
             .append(right))
     }
+
+    fn todo<'a>(&mut self, message: &'a Option<String>, location: &'a SrcSpan) -> Document<'a> {
+        self.tail_position = false;
+        let gleam_error = "todo";
+        let message = message
+            .as_ref()
+            .map(|s| s.as_str())
+            .unwrap_or_else(|| "This has not yet been implemented");
+        let module_name = Document::String(self.module_name.join("_"));
+        let line = location.start;
+
+        println!("{:?}", self);
+        docvec![
+            "throw Object.assign",
```

Comment:
I use `Object.assign` directly here rather than another function call (like `$gleam_error`) to not add any more items to the stacktrace.

Reply:
From this post https://xjamundx.medium.com/custom-javascript-errors-in-es6-aa891b173f87

> One of the common ways to handle this is to append errors with custom properties (which is used heavily in node.js itself.)
> This is a very reliable approach, but has two downsides:
> 
> 1. It can be a lot of lines of code (1 line per extra property)
> 2. Your error instance can’t have any custom methods

I do this here rather than create a custom error, because (1.) is not much of an issue, it's not many lines and they're generated anyway. (2.) I don't have a need for any custom methods on the error. 

To create a custom error would need to be done once per project, so that anyone catching the error could use `instanceof`. To implement that would need us to import a single gleam helpers module which we currently don't have.

So yeah this approach because the downsides aren't very big and a single custom error would be more complexity than it's worth to add.

Reply:
Makes sense, thank you. I would be good to have the extra properties get printed in the console but I can't find a way to do that in JavaScript without generating lots of extra classes.